c/types.go

```diff
@@ -82,27 +100,66 @@ type NotifyStartFn = func(startError)
 // with the supervisor's ChildSpec.
 type ChildSpec struct {
 	name     string
+	tag      ChildTag
```

Comment:
It's been a while, so I have forgotten some things, but I am a bit confused. The comment above leads me to believe that `ChildSpec` is, by definition, a worker. Can you clarify? Maybe the abstraction is getting in the way, and explicit WorkerChildSpec and SupervisorChildSpec might be better than the type? I am really not sure. Just thinking as I type.

Reply:
In practice, this is correct, you see that a sub-tree supervisor is implemented in terms of a [child/worker goroutine](https://github.com/capatazlib/go-capataz/blob/master/s/subtree.go#L32).

I've thought about this, IMO, adding a _type heriarchy_ of `WorkerChildSpec` and `SupervisorChildSpec` enforces us to define an interface implementation for each sub-type (I foresee only two of them), that is similar in all senses except an if statement for sending notifications to the events channel.

To summarize: a `Child` is either be a sub-tree (`Supervisor`) or a leaf (`Worker`), the only difference right now is that we deal with notifications differently [when the `Child` is a `Worker`](https://github.com/capatazlib/go-capataz/pull/26/files#diff-75e644655432015c82ca534acf607b3cR103) and when the `Child` is a `Supervisor` (supervisors are responsible on their own notifications).